src/sign-authorization.ts

```diff
@@ -0,0 +1,127 @@
+import { encode } from '@ethereumjs/rlp';
+import { ecsign, publicToAddress, toBuffer } from '@ethereumjs/util';
+import { bytesToHex } from '@metamask/utils';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+
+import { concatSig, isNullish, recoverPublicKey } from './utils';
+
+/**
+ * The authorization struct as defined in EIP-7702.
+ *
+ * @property chainId - The chain ID or 0 for any chain.
+ * @property contractAddress - The address of the contract being authorized.
+ * @property nonce - The nonce of the signing account (at the time of submission).
+ */
+export type Authorization = [
+  chainId: number,
+  contractAddress: string,
+  nonce: number,
+];
+
+/**
+ * Validates an authorization object to ensure all required parameters are present.
+ *
+ * @param authorization - The authorization object to validate.
+ * @throws {Error} If the authorization object or any of its required parameters are missing.
+ */
+function validateAuthorization(authorization: Authorization) {
```

Comment:
Minor, non-exported functions at the bottom of the file to highlight what is used externally?